server/service/osquery_utils/queries.go

```diff
@@ -1709,19 +1709,19 @@ func directIngestMDMMac(ctx context.Context, logger log.Logger, host *fleet.Host
 			// We should clean this up at some point, but for now we'll just check both.
 			fleetEnrollRef = serverURL.Query().Get("enrollment_reference")
 		}
-		if fleetEnrollRef != "" {
-			if err := ds.SetOrUpdateHostEmailsFromMdmIdpAccounts(ctx, host.ID, fleetEnrollRef); err != nil {
-				if !fleet.IsNotFound(err) {
-					return ctxerr.Wrap(ctx, err, "updating host emails from mdm idp accounts")
-				}
-
-				level.Warn(logger).Log(
-					"component", "service",
-					"method", "directIngestMDMMac",
-					"msg", err.Error(),
-				)
-			}
-		}
+		// if fleetEnrollRef != "" {
```

Comment:
TODO: Confirm we can get rid of this and just rely on the MDM lifecycle events to create/update/delete host email entries for MDM IdP accounts.

Reply:
@gillespi314 was thinking of scenarios like the host deleted from the UI (so we delete `host_mdm` but it gets re-populated afterwards)

Reply:
@roperzh, hmm, I don't think the `host_mdm` table would be an issue since we'd still be saving the `fleet_enroll_ref` there (that's covered by the code above the `if` block). But maybe we do need to think about what happens to the `host_emails` table. If we ever delete entries from `host_emails` where source is `mdm_idp_accounts`, then we'd have a problem and we'd need to be reconciling `host_emails` and `mdm_idp_accounts` somewhere TBD.

Reply:
although we don't delete from `mdm_idp_accounts`, right? it might be fine. My gut feeling is to leave it, but I won't fight for it :D

Reply:
@gillespi314 ah we commented at the same time something similar

Reply:
@roperzh, right, we shouldn't be deleting from `mdm_idp_accounts`.

I don't think we can just leave this as-is because it just catches hosts that have a legacy reference (in the URL reported by osquery). We'd need to add something somewhere that cross-checks host uuids against `mdm_idp_accounts` table and inserts `host_emails` if we have a match in order to catch the delete/reenroll scenario. WDYT?

Reply:
@gillespi314 I'm not sure I fully follow, but If you think it's safe and better to remove this, let's go for it!